src/odemis/dataio/hdf5.py

```diff
@@ -1663,3 +1647,14 @@ def read_thumbnail(filename):
     # TODO: support filename to be a File or Stream
 
     return _thumbFromHDF5(filename)
+
+
+def read_str(s):
+    """
+    Make sure we can read both bytes (HDF5 ascii) and string (HDF5 utf8) metadata
+    That's important because sometimes strings are stored as bytes (ie ascii).
+    This forces both ascii and utf8 to be stored as strings (unicode)
+    """
+    if isinstance(s, bytes):
+        s = s.decode('utf8')
```

Comment:
Please add `, "replace"` argument. That ensures that if some character is not a proper UTF8 bytes, it doesn't totally fail, but still pass to the user the basic content of the string.